Solutions/Ais.Net.Receiver/ReceiverHost.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="Program.cs" company="Endjin">
+// Copyright (c) Endjin. All rights reserved.
+// </copyright>
+
+namespace Ais.Net.Receiver
+{
+    using Microsoft.Extensions.Configuration;
+
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reactive.Linq;
+    using System.Threading.Tasks;
+
+    public class ReceiverHost
+    {
+        private readonly IConfiguration configuration;
+        private readonly IStorageClient storageClient;
+        private NmeaReceiver receiver;
+
+        public ReceiverHost(IConfiguration configuration, IStorageClient storageClient)
+        {
+            this.configuration = configuration;
+            this.storageClient = storageClient;
+        }
+
+        public async Task StartAsync()
+        {
+            this.receiver = new NmeaReceiver(
+                this.configuration[ConfigurationKeys.Host],
```

Comment:
FWIW, I much prefer configuration binding for this sort of thing. If you defined a class with properties for each of these settings you'd have two benefits:

1. you wouldn't have to write all these calls to `Parse` yourself
2. you have a class whose structure tells you exactly what structure of config is required, along with the configuration keys, all in once place (so I don't have to look here to see the expected format, and then in `ConfigurationKeys` to find out the setting name; I'd just have a `public int Host { get; set; }` property that tells me both.

or does `TimeSpan` not work with config binding? Or is it that you particularly wanted these unixy names (all lowercase and dashes) and I'm not sure you can actually do that with config binding (not the dashes anyway).